ext/mysqli/mysqli_nonapi.c

```diff
@@ -769,8 +768,8 @@ static int mysqlnd_zval_array_from_mysqlnd_array(MYSQLND **in_array, zval *out_a
 			MYSQLI_RESOURCE *my_res;
 			mysqli_object *intern = Z_MYSQLI_P(elem);
 			if (!(my_res = (MYSQLI_RESOURCE *)intern->ptr)) {
-		  		php_error_docref(NULL, E_WARNING, "[%d] Couldn't fetch %s", i, ZSTR_VAL(intern->zo.ce->name));
-				continue;
+		  		zend_throw_error(NULL, "%s object is already closed", ZSTR_VAL(intern->zo.ce->name));
+				return FAILURE;
```

Comment:
If we're sure this cannot happen (and I agree with you there), then we can replace this with an assertion.
```
my_res = (MYSQLI_RESOURCE *)intern->ptr;
ZEND_ASSERT(my_res && "Ensure by to_mysqlnd_array");
```

One thing I'm wondering right now is whether we can't cause a crash via mysqlnd_dont_poll_zval_array_from_mysqlnd_array if we provide non-mysqli objects, as that function does not check the object type, unlike the others.